pkg/exchange/coinbase/parse.go

```diff
@@ -0,0 +1,101 @@
+package coinbase
+
+import (
+	"encoding/json"
+	"errors"
+)
+
```

Comment:
It should be 

```
type messageBaseType struct {
	Type string `json:"type"`
         RawData json.RawMessage
}

base := &messageBaseType{}
err := json.Unmarshal(data, &e.RawData)
if err != nil { return err }

switch base.Type {
case "heartbeat":
    var msg HeartbeatMessage
    if err = json.Unmarshal(data, &msg);err!=nil { return err }
    return &msg


}
```

The same logic can refer to https://github.com/c9s/bbgo/blob/74feb928c9da241e8ffd45a06ca77ac4b8076a16/pkg/exchange/bybit/stream.go#L198

Reply:
Done.